draft-bbs-signatures.md

```diff
@@ -633,9 +633,50 @@ Outputs:
 
 Procedure:
 
-1. result = hash_to_field(msg, 1)
+1. If len(dst) > 2^8 - 1 or len(msg) > 2^64 - 1, abort
 
-2. return result
+2. dst_prime = I2OSP(len(dst), 1) || dst
+
+3. msg_prime = I2OSP(len(msg), 8) || msg
+
+4. result = hash_to_scalar(msg_prime || dst_prime, 1)
+
+5. return result
+```
+
+### Hash to scalar
+This operation describes how to hash an arbitrary octet string to `n` scalar values in the multiplicative group of integers mod q. This procedure acts as a helper function, and it is used internally in various places within the operations described in the spec. To map a message to a scalar that would be passed as input to the [Sign](#sign), [Verify](#verify), [spkGen](#spkgen) and [spkVerify](#spkverify) functions, one must use [MapMessageToScalarAsHash](#mapmessagetoscalar) instead.
+
+The `hash_to_scalar` procedure hashes elements using an extendable-output function (XOF). Applications not wishing to use an XOF may use `hash_to_field` defined in Section 5.3 of [@!I-D.irtf-cfrg-hash-to-curve], combined with `expand_message_xmd` defined in Section 5.4.1 of the same document, in place of `hash_to_scalar`. In that case, every element outputted by `hash_to_field` that is equal to 0 MUST be rejected. If that occurs, one should calculate more field elements (using `hash_to_field`), until they get `n` non-zero elements (for example, if there is only one 0 in the output of `hash_to_field(msg, 2)` one must try to calculate `hash_to_field(msg, 3)` etc.).
```

Comment:
I assume whether you are using an XOF vs a conventional HASH alg is a decision for the ciphersuite, how is that captured?

Reply:
Thanks! That's a good point! Added an additional option in the ciphersuites: `hashing_elements_to_scalars`, to capture the two options.

My thinking: When it comes to hash-to-scalar, i don't think it needs to be captured in the ciphersuite, since it will be used by the sign and spkGen operations, i.e., no-need to communicate the use of HASH vs XOF to a verifying/relying party.

For MapMessageToScallarAsHash (that uses hash-to-scalar), again it may not be necessary (since it's use should be agreed in a higher level at some extent e.g., to which messages it applies) but better be careful.